### DISCUSSION TITLE
`whenGuardss01.fs` and `whenGuardss02.fs` look just like `whenGuards01.fs` and `whenGuards02.fs`

### DISCUSSION THREAD

**Discussion by jsoref** [dotnet/fsharp#16185] [Q&A]
**Repro steps**

Provide the steps required to reproduce the problem:

1. `gh repo clone dotnet/fsharp`
2. `cd fsharp`
3. `for a in 1 2; do diff -u tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/whenGuards0$a.fs tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/whenGuardss0$a.fs; done|wc -l`

**Expected behavior**

Some output which signifies why `whenGuardss` files exist -- they appear to be identical to `whenGuards` files.

**Actual behavior**

`0`

**Known workarounds**

Delete the `whenGuardss` files ([`whenGuardss01.fs`](https://github.com/dotnet/fsharp/blob/be6352d1b6325ca93edd8487b19b333a81545201/tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/whenGuardss01.fs) and [`whenGuardss02.fs`](https://github.com/dotnet/fsharp/blob/be6352d1b6325ca93edd8487b19b333a81545201/tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/whenGuardss02.fs)) and references to them.

This is a `whenGuards` reference:
https://github.com/dotnet/fsharp/blob/be6352d1b6325ca93edd8487b19b333a81545201/tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/Expression.fs#L98-L105

This is a `whenGuardss` reference:
https://github.com/dotnet/fsharp/blob/be6352d1b6325ca93edd8487b19b333a81545201/tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/Expression.fs#L116-L123

**Related information**

The files appear to be part of the original commit in this repository, so there isn't any history to explain the reasons they exist
https://github.com/dotnet/fsharp/commits/fb01e0a4858b3f0b24ee115664b4b3255485f268/tests/fsharpqa/Source/Conformance/PatternMatching/Expression/whenGuardss01.fs?browsing_rename_history=true&new_path=tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Expression/whenGuardss01.fs&original_branch=main

**Comment by vzarytovskii** [comment]
It happens that tests are duplicated. Might be testing different things, and changed over years or automigration issue.

Definitely not a bug though.

**Reply by jsoref** [reply]
The files are identical now and were identical in the initial (3900d411aeb2d7740f4e9fd72fce81b3a8f04357) commit -- I'm having a hard time seeing how they could possibly be testing different things.

It's possible they were different before the history of this repository, but they do not appear to have had independent lives since their initial release.
